# Option 1: Heroes of Pymoli

#### Objective: Genderate a report that the purchasing data for the game "Heroes of Pymoli". This will include the Player Count, Purchasing Anaylsis (Total), Gender Demographics, Purchasing Analysis (Gender), Age Dempgraphics, Top Spenders, Most Popular Items and Most Profitable Items.

In [1]:

# Import Dependencies
import pandas as pd
import numpy as np


In [2]:
# Loading csv file
gamingpath = "purchase_data.csv"

# Reading the csv and store into Pandas DataFrame
gamingpath_df = pd.read_csv(gamingpath, encoding="utf-8")
gamingpath_df.head()



Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count

In [3]:
#Calculating total player count
PlayerCount = len(gamingpath_df["SN"].unique())

#Placing data information into summary DataFrame
TotalPlayersTable = pd.DataFrame({"Total Players":[PlayerCount]})
TotalPlayersTable

Total Players
0            576

## Purchasing Analysis (Total)

In [4]:

#Calculating number of unique items, average price, number of purchases, and total revenue
UniqueItemCount = len(gamingpath_df["Item ID"].unique())
AveragePurchasePrice = gamingpath_df["Price"].mean()
TotalNumberPurchases = len(gamingpath_df["Purchase ID"].unique())
TotalPrice = gamingpath_df["Price"].sum()
 
#Placing data information into summary DataFrame
PurchasingAnalysisTable = pd.DataFrame({"Number of Unique Items":[UniqueItemCount], "Average Price":AveragePurchasePrice,
                                        "Number of Purchases":TotalNumberPurchases, "Total Revenue": TotalPrice})

#Formatting average price and total revenue to currency form
PurchasingAnalysisTable["Average Price"] = PurchasingAnalysisTable["Average Price"].map("${:,.2f}".format)
PurchasingAnalysisTable["Total Revenue"] = PurchasingAnalysisTable["Total Revenue"].map("${:,.2f}".format)

#Printing table
PurchasingAnalysisTable


Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780     $2,379.77

## Gender Demographics

In [5]:

#Calculate the percentage and count of all: male, female and other/non-disclosed.
GenderCounts = gamingpath_df["Gender"].value_counts()
GenderPercentage = GenderCounts / PlayerCount * 100

#Placing data information into summary DataFrame
GendersDemSummary = pd.DataFrame({"Percentage Of Players":GenderPercentage,"Total Count":GenderCounts})

#Formatting percentage of players to 2 decimal form
GendersDemSummary["Percentage Of Players"] = GendersDemSummary["Percentage Of Players"].map("{:,.2f}".format)

#Print dataframe
GendersDemSummary



Percentage Of Players  Total Count
Male                                 113.19          652
Female                                19.62          113
Other / Non-Disclosed                  2.60           15

## Purchasing Analysis (Gender)

In [6]:

# Run basic calculations
GenderCounter = gamingpath_df.groupby(["Gender"]).count()["Price"].rename("Purchase Count")
GenderPurchaseAvg = gamingpath_df.groupby(["Gender"]).mean()["Price"].rename("Average Purchase Price")
GenderTotalPur = gamingpath_df.groupby(["Gender"]).sum()["Price"].rename("Total Purchase Value")


# Calculate Normalized Purchasing
GenderNormTotal = GenderTotalPur / GendersDemSummary["Total Count"]

#Placing data information into summary DataFrame
GenderPurAnalysis = pd.DataFrame({"Purchase Count": GenderCounter, "Average Purchase Price": GenderPurchaseAvg, 
                                  "Total Purchase Value": GenderTotalPur, "Normalized Totals": GenderNormTotal})

#Formatting all needed values to currency form
GenderPurAnalysis["Average Purchase Price"] = GenderPurAnalysis["Average Purchase Price"].map("${:,.2f}".format)
GenderPurAnalysis["Total Purchase Value"] = GenderPurAnalysis["Total Purchase Value"].map("${:,.2f}".format)
GenderPurAnalysis ["Purchase Count"] = GenderPurAnalysis["Purchase Count"]
GenderPurAnalysis["Normalized Totals"] = GenderPurAnalysis["Normalized Totals"].map("${:,.2f}".format)

#Print dataframe
GenderPurAnalysis



Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Normalized Totals  
Gender                                                        
Female                             $361.94             $3.20  
Male                             $1,967.64             $3.02  
Other / Non-Disclosed               $50.19             $3.35

## Age Demographics

In [7]:
# Create bins in which to place values based upon player's ages
age_bins = [0, 9.90, 14.90, 19.90, 24.90, 29.90, 34.90, 39.90, 99999]

# Create labels for these bins
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]



In [8]:
# Slice the data and place it into bins
pd.cut(gamingpath_df["Age"], age_bins, labels=group_names).head()

0    20-24
1      40+
2    20-24
3    20-24
4    20-24
Name: Age, dtype: category
Categories (8, object): [<10 < 10-14 < 15-19 < 20-24 < 25-29 < 30-34 < 35-39 < 40+]

In [9]:
# Place data series into a new column inside DataFrame
gamingpath_df[" "] = pd.cut(gamingpath_df["Age"], age_bins, labels=group_names)
gamingpath_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price         
0  Extraction, Quickblade Of Trembling Hands   3.53  20-24  
1                          Frenzied Scimitar   1.56    40+  
2                               Final Critic   4.88  20-24  
3                                Blindscythe   3.27  20-24  
4                                       Fury   1.44  20-24

In [10]:
#Calculating percentage and total players per age group
AgeCounter = gamingpath_df.groupby([" "]).count()["Age"].rename("Total Count")
AgePercent = AgeCounter / PlayerCount * 100

#Creating summary dataframe
AgeDemoTable = pd.DataFrame({"Percentage of Players": AgePercent, "Total Count": AgeCounter})

# Formating to 2 place decimals
AgeDemoTable["Percentage of Players"] = AgeDemoTable["Percentage of Players"].map("{:,.2f}".format)

#Print table
AgeDemoTable





Percentage of Players  Total Count
                                        
<10                    3.99           23
10-14                  4.86           28
15-19                 23.61          136
20-24                 63.37          365
25-29                 17.53          101
30-34                 12.67           73
35-39                  7.12           41
40+                    2.26           13

In [11]:
#Binning the purchase_data dataframe by age.
# Run basic calculations
AgePurCounter = gamingpath_df.groupby([" "]).count()["Price"].rename("Purchase Count")
AgePurchaseAvg = gamingpath_df.groupby([" "]).mean()["Price"].rename("Average Purchase Price")
AgeTotalPur = gamingpath_df.groupby([" "]).sum()["Price"].rename("Total Purchase Value")


# Calculate Normalized Purchasing
AgeNormTotal = AgeTotalPur / AgeCounter

# Convert to DataFrame
AgePurAnalysis = pd.DataFrame({"Purchase Count": AgePurCounter, "Average Purchase Price": AgePurchaseAvg, 
                                  "Total Purchase Value": AgeTotalPur, "Normalized Totals": AgeNormTotal})

#Formatting all needed values to currency form
AgePurAnalysis["Average Purchase Price"] = AgePurAnalysis["Average Purchase Price"].map("${:,.2f}".format)
AgePurAnalysis["Total Purchase Value"] = AgePurAnalysis["Total Purchase Value"].map("${:,.2f}".format)
AgePurAnalysis ["Purchase Count"] = AgePurAnalysis["Purchase Count"]
AgePurAnalysis["Normalized Totals"] = AgePurAnalysis["Normalized Totals"].map("${:,.2f}".format)
#Print data frame
AgePurAnalysis

Purchase Count Average Purchase Price Total Purchase Value  \
                                                                    
<10                23                  $3.35               $77.13   
10-14              28                  $2.96               $82.78   
15-19             136                  $3.04              $412.89   
20-24             365                  $3.05            $1,114.06   
25-29             101                  $2.90              $293.00   
30-34              73                  $2.93              $214.00   
35-39              41                  $3.60              $147.67   
40+                13                  $2.94               $38.24   

      Normalized Totals  
                         
<10               $3.35  
10-14             $2.96  
15-19             $3.04  
20-24             $3.05  
25-29             $2.90  
30-34             $2.93  
35-39             $3.60  
40+               $2.94

## Top Spenders

In [12]:

#Finding top 5 spenders in game by total purchase value, then list table

#Group data by each player
PlayerTopCount = gamingpath_df.groupby(["SN"]).count()["Price"].rename("Purchase Count")
PlayerPurchaseAvg = gamingpath_df.groupby(["SN"]).mean()["Price"].rename("Average Purchase Price")
PlayerTotalPur = gamingpath_df.groupby(["SN"]).sum()["Price"].rename("Total Purchase Value")

# Convert to DataFrame
TopPlayers = pd.DataFrame({"Purchase Count": PlayerTopCount,"Average Purchase Price": PlayerPurchaseAvg, 
                           "Total Purchase Value":PlayerTotalPur})
#Sort data highest to lowest by total purchase value 
TopPlayers = TopPlayers.sort_values(by='Total Purchase Value', ascending=False)

#Formatting all needed values to currency form
TopPlayers["Average Purchase Price"] = TopPlayers["Average Purchase Price"].map("${:,.2f}".format)
TopPlayers["Total Purchase Value"] = TopPlayers["Total Purchase Value"].map("${:,.2f}".format)

#Print dataframe only displaying first 5 rows.
TopPlayers.head(5)


Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

## Most Popular Items

In [13]:

#Finding top 5 most popular items in the game by purchase count

#Group data by item ID
PopularTopCount = gamingpath_df.groupby(["Item ID","Item Name"]).count()["Price"].rename("Purchase Count")
PopularPurchaseAvg = gamingpath_df.groupby(["Item ID","Item Name"]).mean()["Price"].rename("Item Price")
PopularTotalPur = gamingpath_df.groupby(["Item ID","Item Name"]).sum()["Price"].rename("Total Purchase Value")

# Convert to DataFrame
PopularItems = pd.DataFrame({"Purchase Count": PopularTopCount,"Item Price": PopularPurchaseAvg, 
                             "Total Purchase Value":PopularTotalPur})
#Sort data highest to lowest by total purchase value
PopularItems = PopularItems.sort_values(by='Purchase Count', ascending=False)
#Formatting all needed values to currency form
PopularItems["Total Purchase Value"] = PopularItems["Total Purchase Value"].map("${:,.2f}".format)
PopularItems["Item Price"] = PopularItems["Item Price"].map("${:,.2f}".format)

#Print dataframe only displaying first 5 rows.
PopularItems.head(5)

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16

## Most Profitable Items

In [14]:

#Finding top 5 most profitable items in the game by total purchase value

#Group data by Item ID & Item Name
ProfitableTopCount = gamingpath_df.groupby(["Item ID","Item Name"]).count()["Price"].rename("Purchase Count")
ProfitablePurchaseAvg = gamingpath_df.groupby(["Item ID","Item Name"]).mean()["Price"].rename("Item Price")
ProfitableTotalPur = gamingpath_df.groupby(["Item ID","Item Name"]).sum()["Price"].rename("Total Purchase Value")

# Convert to DataFrame
ProfitableItems = pd.DataFrame({"Purchase Count": ProfitableTopCount,"Item Price": ProfitablePurchaseAvg, 
                           "Total Purchase Value":ProfitableTotalPur})
#Sort data highest to lowest by total purchase value
ProfitableItems = ProfitableItems.sort_values(by='Total Purchase Value', ascending=False)
#Formatting all needed values to currency form
ProfitableItems["Total Purchase Value"] = ProfitableItems["Total Purchase Value"].map("${:,.2f}".format)
ProfitableItems["Item Price"] = ProfitableItems["Item Price"].map("${:,.2f}".format)

#Print dataframe only displaying first 5 rows.
ProfitableItems.head(5)

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
92      Final Critic                                      $4.88   
103     Singed Scalpel                                    $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
92      Final Critic                                               $39.04  
103     Singed Scalpel                                             $34.80

# Heroes of Pymoli – Observable Trends

##### Heroes of Pymoli game purchasing analysis has been broken down into 8 categories: Player Count, Purchasing Analysis (Total), Gender Demographics, Purchasing Analysis (Gender), Age Demographics, Top Spenders, Most Popular Items and Most Profitable Items.
##### According to the data the following 3 trends can be observed:
##### 1.	Heroes of Pymoli has 576 total players currently registered. These users have made 780 purchases to enhance their gaming experience. Even though males represent the highest percentage of players 113.19%. Females and Other/Non-Disclosed users on average tend to purchase the more expensive items of 3.20 USD (Female) and 3.35 USD (Other/Non-Disclosed).
#### 2.	Amongst the age demographics of 20–24 represent 63.37% of registered users with an average purchase of  3.05 USD. However, users between the ages of 35-39 on average tend to purchase higher value items at 3.60 USD.
#### 3.	The Top Spender will make between 3 to 5 purchases with an average purchase price of 3.40 USD to 4.37 USD. With the most popular and profitable purchased item being the “Oathbreaker, Last Hope of the Breaking Storm”.


